torch/csrc/jit/serialization/inlined_callstack_serialization.cpp

```diff
@@ -0,0 +1,212 @@
+#include <torch/csrc/jit/api/compilation_unit.h>
+#include <torch/csrc/jit/serialization/inlined_callstack_serialization.h>
+#include <torch/csrc/jit/serialization/pickle.h>
+
+namespace torch {
+namespace jit {
+
+c10::IValue InlinedCallStackSerializer::serialize(
+    const InlinedCallStackPtr& cs_ptr,
+    const SourceRangeTagMap& source_range_tags) {
+  if (!cs_ptr) {
+    return c10::IValue();
+  }
+  auto cs_it = serialized_inlined_callstack_.find(cs_ptr);
+  if (cs_it != serialized_inlined_callstack_.end()) {
+    return cs_it->second;
+  }
+  // Inlined callstack pointer is serialized as tuple of 3 elements
+  // {IValue(module_instance_info), source_range_tag, IValue(InlinedCallStack)}
+  std::vector<c10::IValue> elements;
+  elements.reserve(3);
+  elements.emplace_back(
+      serialize_module_instance_info(cs_ptr->module_instance()));
```

Comment:
Do we expect to serialize the same module_instance_info many times?  Should we cache the serialized values so the pickler can re-use them?

Reply:
Thanks for the catch David. Yes I do expect that the same module instance will appear in multiple inlined callstacks. Let me add a todo here and file a task to track this. I will add this in a separate PR.